sources/packages/web/src/components/layouts/student/StudentPageContainer.vue

```diff
@@ -0,0 +1,45 @@
+<template>
+  <full-page-container
+    :layout-template="layoutTemplate"
+    :full-width="fullWidth"
+  >
+    <template #header>
+      <slot name="header"></slot>
+    </template>
+    <template #alerts>
+      <restriction-banner v-if="!hideRestriction" />
+      <CheckValidSINBanner />
+    </template>
+    <slot name="content"></slot>
+  </full-page-container>
+</template>
+<script lang="ts">
+import RestrictionBanner from "@/views/student/RestrictionBanner.vue";
+import CheckValidSINBanner from "@/views/student/CheckValidSINBanner.vue";
+enum LayoutTemplates {
```

Comment:
Should we duplicate this? Maybe it can be moved to the contracts folder.

Reply:
This is an example where the enum is present but we use the strings to set the values. So the enum was created for the internal component organization, but the component consumer can provide the value as strings, the same behavior for other Vue components in general (I believe).

Reply:
I believe, we can have it in the template rather having it in the consumer everytime.

Reply:
mm... Iam okay with both

Reply:
Trying to settle this, I believe that we can have both ways we just do not need to enforce it.